### PR TITLE
Improve readability of "Daily Activity" API endpoint

### PR DESCRIPTION
**Pull request by ishaanshah**
# Problem
The "Daily Activity" API endpoint is confusing and hard to understand.

# Solution
Grouping hours of each day into a single list solves the issue.

### CODE REVIEW

File: listenbrainz/webserver/views/stats_api.py [hunk 1]
```diff
@@ -477,7 +474,23 @@ def get_daily_activity(user_name: str):
...
-    daily_activity = [x.dict() for x in getattr(stats, stats_range).daily_activity]
+    daily_activity_unprocessed = [x.dict() for x in getattr(stats, stats_range).daily_activity]
+    daily_activity = {calendar.day_name[day]: [{"hour": hour, "listen_count": 0} for hour in range(0, 24)] for day in range(0, 7)}
+
+    for day, day_data in daily_activity.items():
+        for hour_data in day_data:
+            hour = hour_data["hour"]
+
+            found = False
+            for entry in daily_activity_unprocessed:
```
Comment (paramsingh) [inline listenbrainz/webserver/views/stats_api.py:485]: you can use a `for...else` here to get rid of the `found` variable. (https://book.pythontips.com/en/latest/for_-_else.html)

File: listenbrainz/webserver/views/stats_api.py [hunk 2]
```diff
@@ -530,6 +543,8 @@ def _get_entity_list(
     entity: str,
     offset: int,
     count: int,
+
```
Comment (paramsingh) [inline listenbrainz/webserver/views/stats_api.py:546]: could we remove these spaces?

File: listenbrainz/webserver/views/stats_api.py (hunk 1 above)
Comment (ishaanshah) [reply listenbrainz/webserver/views/stats_api.py:484]: Neat trick!

File: listenbrainz/webserver/views/stats_api.py (hunk 2 above)
Comment (ishaanshah) [reply listenbrainz/webserver/views/stats_api.py]: Fixed it.

**Pull request merged**